solidity/ecdsa/test/WalletRegistry.Slashing.test.ts

```diff
@@ -0,0 +1,152 @@
+              walletID,
+              membersIDs
+            )
+        ).to.be.revertedWith("Caller is not the Wallet Owner")
+      })
+    })
+
+    context("when called by the wallet owner", () => {
+      context("when the passed wallet members identifiers are invalid", () => {
+        it("should revert", async () => {
+          const corruptedMembersIDs = membersIDs.slice().reverse()
+          await expect(
+            walletRegistry
+              .connect(walletOwner.wallet)
+              .seize(
+                amountToSlash,
+                rewardMultiplier,
+                thirdParty.address,
+                walletID,
+                corruptedMembersIDs
+              )
+          ).to.be.revertedWith("Invalid wallet members identifiers")
+        })
+      })
+
+      context("when the passed wallet members identifiers are valid", () => {
+        let notifierBalanceBefore
+        let notifierBalanceAfter
+
+        before(async () => {
+          await createSnapshot()
+
+          notifierBalanceBefore = await tToken.balanceOf(thirdParty.address)
+          await walletRegistry
+            .connect(walletOwner.wallet)
+            .seize(
+              amountToSlash,
+              rewardMultiplier,
+              thirdParty.address,
+              walletID,
+              membersIDs
+            )
+          notifierBalanceAfter = await tToken.balanceOf(thirdParty.address)
+        })
+
+        after(async () => {
+          await restoreSnapshot()
+        })
+
+        it("should slash all group members", async () => {
+          expect(await staking.getSlashingQueueLength()).to.equal(
+            constants.groupSize
+          )
+        })
+
+        it("should slash with correct amounts", async () => {
+          for (let i = 0; i < constants.groupSize; i++) {
+            const slashing = await staking.slashingQueue(i)
+            expect(slashing.amount).to.equal(amountToSlash)
+          }
+        })
+
+        it("should slash correct staking providers", async () => {
+          for (let i = 0; i < constants.groupSize; i++) {
+            const slashing = await staking.slashingQueue(i)
+            const expectedStakingProvider =
+              await walletRegistry.operatorToStakingProvider(
+                membersAddresses[i]
+              )
+
+            expect(slashing.stakingProvider).to.equal(expectedStakingProvider)
+          }
+        })
+
+        it("should send correct reward to notifier", async () => {
+          // reward multiplier is in % so we first multiply and then divide by
+          // 100 to get the actual number
+          const perMemberReward = constants.tokenStakingNotificationReward
+            .mul(rewardMultiplier)
+            .div(100)
+
+          const receivedReward = notifierBalanceAfter.sub(notifierBalanceBefore)
+
+          expect(receivedReward).to.equal(
+            perMemberReward.mul(constants.groupSize)
+          )
+        })
+      })
```

Comment:
Can we add a test case to validate that the transaction reverts if `staking.seize` fails?

Reply:
Yes and no - my favourite answer. See #2870.

Reply:
Maybe it would be worth adding at least a placeholder so we know that when the referenced issue is fixed we need to implement a test?